Build fails if git is configured to show signatures by default

**Affects versions :**
 - OS: all
 - Which version of libgtk do you use: 3.24.24
 - Version of Xournal++: `master` branch
 - Installation method: build using source from Git tree

**Describe the bug**
If `git log` is configured to show PGP signatures by default (e.g. with git aliases), then the build will fail because certain commands used to print the current commit ID contain extra output.

**To Reproduce**
Steps to reproduce the behavior:
1. run `git config log.showSignature true` to enable PGP signatures in `git log`
2. build Xornal++

**Expected behavior**
The build should ignore/override the show-signature option and continue as normal.

**Additional context**
Here's the output from the build. As you can see, `GIT_COMMIT_ID`, which should have the commit ID, instead has signature information from GPG:
```text
[ 21%] Building CXX object src/util/CMakeFiles/util.dir/audio/PortAudioConsumer.cpp.o
In file included from /home/kian/.cache/yay/xournalpp-git/src/xournalpp/src/undo/UndoAction.h:16,
                 from /home/kian/.cache/yay/xournalpp-git/src/xournalpp/src/undo/UndoRedoHandler.h:20,
                 from /home/kian/.cache/yay/xournalpp-git/src/xournalpp/src/control/Control.h:26,
                 from /home/kian/.cache/yay/xournalpp-git/src/xournalpp/src/util/audio/AudioPlayer.h:17,
                 from /home/kian/.cache/yay/xournalpp-git/src/xournalpp/src/util/audio/AudioPlayer.cpp:1:
/home/kian/.cache/yay/xournalpp-git/src/xournalpp/build/src/config.h:77:32: warning: missing terminating " character
   77 | constexpr auto GIT_COMMIT_ID = "gpg: Signature made Sat 05 Dec 2020 06:30:58 AM PST
      |                                ^
/home/kian/.cache/yay/xournalpp-git/src/xournalpp/build/src/config.h:77:32: error: missing terminating " character
   77 | constexpr auto GIT_COMMIT_ID = "gpg: Signature made Sat 05 Dec 2020 06:30:58 AM PST
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/kian/.cache/yay/xournalpp-git/src/xournalpp/build/src/config.h:79:67: error: stray ‘@’ in program
   79 | gpg: Good signature from GitHub (web-flow commit signing) <noreply@github.com> [full]
      |                                                                   ^
/home/kian/.cache/yay/xournalpp-git/src/xournalpp/build/src/config.h:80:9: warning: missing terminating " character
   80 | 5cf01b77";
      |         ^
/home/kian/.cache/yay/xournalpp-git/src/xournalpp/build/src/config.h:80:9: error: missing terminating " character
   80 | 5cf01b77";
      |         ^~
In file included from /home/kian/.cache/yay/xournalpp-git/src/xournalpp/src/undo/UndoAction.h:16,
                 from /home/kian/.cache/yay/xournalpp-git/src/xournalpp/src/undo/UndoRedoHandler.h:20,
                 from /home/kian/.cache/yay/xournalpp-git/src/xournalpp/src/control/Control.h:26,
                 from /home/kian/.cache/yay/xournalpp-git/src/xournalpp/src/util/audio/AudioPlayer.h:17,
                 from /home/kian/.cache/yay/xournalpp-git/src/xournalpp/src/util/audio/AudioPlayer.cpp:1:
/home/kian/.cache/yay/xournalpp-git/src/xournalpp/build/src/config.h:78:1: error: ‘gpg’ was not declared in this scope
   78 | gpg:                using RSA key 4AEE18F83AFDEB23
      | ^~~
/home/kian/.cache/yay/xournalpp-git/src/xournalpp/build/src/config.h:77:16: warning: ‘GIT_COMMIT_ID’ defined but not used [-Wunused-variable]
   77 | constexpr auto GIT_COMMIT_ID = "gpg: Signature made Sat 05 Dec 2020 06:30:58 AM PST
      |                ^~~~~~~~~~~~~
make[2]: *** [src/util/CMakeFiles/util.dir/build.make:251: src/util/CMakeFiles/util.dir/audio/AudioPlayer.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /home/kian/.cache/yay/xournalpp-git/src/xournalpp/src/undo/UndoAction.h:16,
                 from /home/kian/.cache/yay/xournalpp-git/src/xournalpp/src/undo/UndoRedoHandler.h:20,
                 from /home/kian/.cache/yay/xournalpp-git/src/xournalpp/src/control/Control.h:26,
                 from /home/kian/.cache/yay/xournalpp-git/src/xournalpp/src/util/audio/AudioPlayer.h:17,
                 from /home/kian/.cache/yay/xournalpp-git/src/xournalpp/src/util/audio/PortAudioConsumer.cpp:5:
/home/kian/.cache/yay/xournalpp-git/src/xournalpp/build/src/config.h:77:32: warning: missing terminating " character
   77 | constexpr auto GIT_COMMIT_ID = "gpg: Signature made Sat 05 Dec 2020 06:30:58 AM PST
      |                                ^
/home/kian/.cache/yay/xournalpp-git/src/xournalpp/build/src/config.h:77:32: error: missing terminating " character
   77 | constexpr auto GIT_COMMIT_ID = "gpg: Signature made Sat 05 Dec 2020 06:30:58 AM PST
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/kian/.cache/yay/xournalpp-git/src/xournalpp/build/src/config.h:79:67: error: stray ‘@’ in program
   79 | gpg: Good signature from GitHub (web-flow commit signing) <noreply@github.com> [full]
      |                                                                   ^
/home/kian/.cache/yay/xournalpp-git/src/xournalpp/build/src/config.h:80:9: warning: missing terminating " character
   80 | 5cf01b77";
      |         ^
/home/kian/.cache/yay/xournalpp-git/src/xournalpp/build/src/config.h:80:9: error: missing terminating " character
   80 | 5cf01b77";
      |         ^~
/home/kian/.cache/yay/xournalpp-git/src/xournalpp/build/src/config.h:78:1: error: ‘gpg’ was not declared in this scope
   78 | gpg:                using RSA key 4AEE18F83AFDEB23
      | ^~~
/home/kian/.cache/yay/xournalpp-git/src/xournalpp/build/src/config.h:77:16: warning: ‘GIT_COMMIT_ID’ defined but not used [-Wunused-variable]
   77 | constexpr auto GIT_COMMIT_ID = "gpg: Signature made Sat 05 Dec 2020 06:30:58 AM PST
      |                ^~~~~~~~~~~~~
make[2]: *** [src/util/CMakeFiles/util.dir/build.make:290: src/util/CMakeFiles/util.dir/audio/PortAudioConsumer.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:423: src/util/CMakeFiles/util.dir/all] Error 2
make: *** [Makefile:182: all] Error 2
```

Here's a patch that fixes the issue:
```diff
diff --git a/cmake/include/Version.cmake b/cmake/include/Version.cmake
index ab40470f..b3138309 100644
--- a/cmake/include/Version.cmake
+++ b/cmake/include/Version.cmake
@@ -46,7 +46,7 @@ function(core_find_git_rev stamp)
                         WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
         endif()
         # get HEAD commit SHA-1
-        execute_process(COMMAND ${GIT_EXECUTABLE} log -n 1 --pretty=format:"%h" HEAD
+        execute_process(COMMAND ${GIT_EXECUTABLE} log --no-show-signature -n 1 --pretty=format:"%h" HEAD
                         OUTPUT_VARIABLE HASH
                         WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
         string(REPLACE "\"" "" HASH ${HASH})
@@ -56,7 +56,7 @@ function(core_find_git_rev stamp)
         endif()
 
       # get HEAD commit date
-      execute_process(COMMAND ${GIT_EXECUTABLE} log -1 --pretty=format:"%cd" --date=short HEAD
+      execute_process(COMMAND ${GIT_EXECUTABLE} log --no-show-signature -1 --pretty=format:"%cd" --date=short HEAD
                       OUTPUT_VARIABLE DATE
                       WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
       string(REPLACE "\"" "" DATE ${DATE})
```

--- a/cmake/include/Version.cmake
+++ b/cmake/include/Version.cmake
@@ -46,7 +46,7 @@ function(core_find_git_rev stamp)
                         WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
         endif()
         # get HEAD commit SHA-1
-        execute_process(COMMAND ${GIT_EXECUTABLE} log -n 1 --pretty=format:"%h" HEAD
+        execute_process(COMMAND ${GIT_EXECUTABLE} log --no-show-signature -n 1 --pretty=format:"%h" HEAD
                         OUTPUT_VARIABLE HASH
                         WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
         string(REPLACE "\"" "" HASH ${HASH})
@@ -56,7 +56,7 @@ function(core_find_git_rev stamp)
         endif()
 
       # get HEAD commit date
-      execute_process(COMMAND ${GIT_EXECUTABLE} log -1 --pretty=format:"%cd" --date=short HEAD
+      execute_process(COMMAND ${GIT_EXECUTABLE} log --no-show-signature -1 --pretty=format:"%cd" --date=short HEAD
                       OUTPUT_VARIABLE DATE
                       WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
       string(REPLACE "\"" "" DATE ${DATE})

